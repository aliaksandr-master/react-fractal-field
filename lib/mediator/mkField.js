export const mkMeta = () => ({
  valid: true,
  touched: false,

  active: false,
  activated: false,

  submitted: false,
  submitting: false,
  submitFailed: false,
  submitSuccess: false,
  submittedTimes: 0,

  error: null,
  errorsIn: [],
  hasException: false,
  hasFormatException: false,
  hasNormalizeException: false,
  hasValidateException: false
});

export default ({
  id,
  complexValueType = null,
  parentID = null,
  autoClean = false,
  shareMeta = true,
  shareValue = true,
  value = undefined,
  name = null
}) => ({
  // permanent
  id,
  parentID,
  autoClean,
  shareMeta,
  shareValue,
  complexValueType, // if not null component might have named children

  // free transform
  name,

  value,

  meta: mkMeta(),

  children: [],
  version: 0,
  metaVersion: 0,
  valueVersion: 0
});
