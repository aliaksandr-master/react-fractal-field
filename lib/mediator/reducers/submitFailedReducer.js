import { updateFieldMeta, fieldReducer } from '../stateHelpers';



export default () => fieldReducer((state, field) => {
  state = updateFieldMeta(state, field, {
    submitting: false,
    submitted: true,
    submittedTimes: field.meta.submittedTimes + 1,
    submitSuccess: false,
    submitFailed: true
  });

  return state;
});
