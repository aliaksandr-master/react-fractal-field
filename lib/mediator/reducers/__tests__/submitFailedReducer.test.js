/*eslint-env jest*/
import submitFailedReducer from '../submitFailedReducer';
import mkField from '../../mkField';


test('submitFailedReducer', () => {
  expect(submitFailedReducer()({}, { type: 'some', fieldID: 123 })).toMatchSnapshot();

  expect(submitFailedReducer()({ [123]: mkField({ id: 123 }) }, { type: 'some', fieldID: 123 })).toMatchSnapshot();
});
