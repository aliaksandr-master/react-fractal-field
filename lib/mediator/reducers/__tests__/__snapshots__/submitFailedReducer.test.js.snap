// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitFailedReducer 1`] = `Object {}`;

exports[`submitFailedReducer 2`] = `
Object {
  "123": Object {
    "autoClean": false,
    "children": Array [],
    "complexValueType": null,
    "id": 123,
    "meta": Object {
      "activated": false,
      "active": false,
      "error": null,
      "errorsIn": Array [],
      "hasException": false,
      "hasFormatException": false,
      "hasNormalizeException": false,
      "hasValidateException": false,
      "submitFailed": true,
      "submitSuccess": false,
      "submitted": true,
      "submittedTimes": 1,
      "submitting": false,
      "touched": false,
      "valid": true,
    },
    "metaVersion": 1,
    "name": null,
    "parentID": null,
    "shareMeta": true,
    "shareValue": true,
    "value": undefined,
    "valueVersion": 0,
    "version": 1,
  },
}
`;
